### PR TITLE
Adding space membership endpoint

### PR DESCRIPTION
**Pull request by mariobodemann**
https://www.contentful.com/developers/docs/references/content-management-api/#/reference/space-memberships

### CODE REVIEW

File: src/main/java/com/contentful/java/cma/ModuleSpaceMemberships.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+  /**
+   * Delete the given membership instance.
+   * <p>
+   * Please make sure that the instance provided is fetched from Contentful. Otherwise you will
+   * get an exception thrown.
+   *
+   * @param spaceId    the id of the space to be used.
+   * @param membership the membership fetched from contentful, updated by caller, to be deleted.
+   * @return an empty string on success.
```
Comment (loudmouth) [inline src/main/java/com/contentful/java/cma/ModuleSpaceMemberships.java]: Should be returns an integer status code or something of that sort

File: src/main/java/com/contentful/java/cma/ModuleSpaceMemberships.java [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+     *
+     * @param spaceId the space identifier identifying the space.
+     * @return a callback for the array fetched.
+     * @throws IllegalArgumentException if spaceId is null.
+     * @see ModuleSpaceMemberships#fetchAll(String)
+     */
+    public CMACallback<CMAArray<CMASpaceMembership>> fetchAll(
+        final String spaceId,
+        final CMACallback<CMAArray<CMASpaceMembership>> callback
+    ) {
```
Comment (loudmouth) [inline src/main/java/com/contentful/java/cma/ModuleSpaceMemberships.java]: to me, this looks a bit weird 😳

File: src/main/java/com/contentful/java/cma/ServiceSpaceMemberships.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import retrofit2.http.DELETE;
+import retrofit2.http.GET;
+import retrofit2.http.Header;
+import retrofit2.http.POST;
+import retrofit2.http.PUT;
+import retrofit2.http.Path;
+import rx.Observable;
+
+/**
+ * Service class to define the rest interface to Contentful.
```
Comment (loudmouth) [inline src/main/java/com/contentful/java/cma/ServiceSpaceMemberships.java]: REST

File: src/main/java/com/contentful/java/cma/model/CMASpaceMembership.java [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+  /**
+   * Replace all roles with the given argument.
+   *
+   * @param roles the roles to be used.
+   * @return this instance of membership for ease of chaining.
+   * @throws IllegalArgumentException if roles is null.
+   * @throws IllegalArgumentException if roles does not contain any elements.
+   */
+  public CMASpaceMembership setRoles(CMALink... roles) {
```
Comment (loudmouth) [inline src/main/java/com/contentful/java/cma/model/CMASpaceMembership.java]: variadic's for the win!

File: src/main/java/com/contentful/java/cma/ModuleSpaceMemberships.java (hunk 1 above)
Comment (mariobodemann) [inline src/main/java/com/contentful/java/cma/ModuleSpaceMemberships.java]: Yes, that is a great spot! I forgot to update after some iteration the documentation! (Documentation driven developement for the win!)

File: src/main/java/com/contentful/java/cma/ModuleSpaceMemberships.java (hunk 2 above)
Comment (mariobodemann) [inline src/main/java/com/contentful/java/cma/ModuleSpaceMemberships.java]: Another great spot. It is incosistent with the rest of the sdk, so I'll move them up.

File: src/main/java/com/contentful/java/cma/model/CMASpaceMembership.java (hunk 4 above)
Comment (mariobodemann) [inline src/main/java/com/contentful/java/cma/model/CMASpaceMembership.java]: [Varargs](https://docs.oracle.com/javase/1.5.0/docs/guide/language/varargs.html) in java

**Pull request merged**